src/useBusState.ts

```diff
@@ -0,0 +1,19 @@
+import { useEffect, useState, Dispatch, SetStateAction, useLayoutEffect } from "react";
+import { useBus } from "./react";
+import { BusEvent } from "./types";
+
+export function useBusState<E extends BusEvent = BusEvent>(
+  initState: E["payload"] | undefined,
+  event: E["type"]
+): [ E["payload"], Dispatch<SetStateAction<E["payload"]>>] {
```

Comment:
I am wondering if this complicates things when we don't return an array in `useBusReducer`? Also this dispatch will not trigger the event. Could that be a mistake?

If we return a setter function I would argue it should publish an event and we should return a dispatch function from useBusReducer. Without doing this you are allowing to change the local state without notifying of other parts of the app that rely on that state which will mean you can get into situations where state is inconsistent.

Returning a setter function means that the React adaptor is a duplex connection allowing multiple ways to publish events. I kind of think this makes things a little complex as it means we have multiple ways to publish events that need to be learnt by others on a dev team. On the other hand it could be a more concise way of working with publishing events.

My preference is to simply return the state and if you need to update the state you use a "normal" event.

It would be nice to pass in a predicate, an event type or an eventDefinition but for now we can stick to event type to get the first cut merged.